pkg/resource/deploy/source_eval.go

```diff
@@ -1054,9 +1054,10 @@ func (rm *resmon) RegisterResource(ctx context.Context,
 	}
 
 	propertyDependencies := make(map[resource.PropertyKey][]resource.URN)
-	if len(req.GetPropertyDependencies()) == 0 {
+	if len(req.GetPropertyDependencies()) == 0 && !remote {
 		// If this request did not specify property dependencies, treat each property as depending on every resource
-		// in the request's dependency list.
+		// in the request's dependency list. We don't need to do this when remote is true, because all clients that
+		// support remote already support passing property dependencies, so there's no need to backfill here.
```

Comment:
I'm assuming we are ok with telling people that they need to upgrade their 3.1.0 SDKs here.

Reply:
What's special about 3.1.0 SDKs and why would we need to tell them to upgrade?

Backfilling the property dependencies like this is only an issue for remote components when you have a property that is plain and isn't supposed to be able to have any dependencies associated with it. For register resource calls where remote is false, it doesn't matter.

We added support for remote components in the Go SDK in v2.13.0. Any version of the Go SDK v2.13.0 or higher passing `remote == true` will get this fix when running against a CLI that has this engine change.

Reply:
Sorry for the unclear comment. 3.1.0 is an arbitrary early version number I picked. What I meant was:

From the comment, it sounds like their used to be clients that supported remote but didn't support passing property dependencies. It seems like this will break them. I'm assuming that this break is acceptable because we can tell them to upgrade, and everything should start working again.

Reply:
It won't break anyone. Users running Go SDK version < v2.13.0 aren't passing `remote`, so there's no behavior change.